secure_api/merkle_tree/MerkleTree.Low.Datastructures.fst

```diff
@@ -277,15 +278,15 @@ let hash_vec_r_alloc #_ hsz r =
 #pop-options
 
 val hash_vec_r_free:
-  #hsz:Ghost.erased hash_size_t ->
-  v:hash_vec ->
+  #hsz':Ghost.erased hash_size_t ->
+  hsz:hash_size_t { hsz == Ghost.reveal hsz' } ->
+  v:hash_vec #hsz ->
   HST.ST unit
     (requires (fun h0 -> hash_vec_r_inv h0 v))
     (ensures (fun h0 _ h1 ->
       modifies (loc_all_regions_from false (hash_vec_region_of #hsz v)) h0 h1))
-let hash_vec_r_free #_ v =
-  // RV.free v
-  V.free v
+let hash_vec_r_free #_ hsz v =
+  RV.free v
```

Comment:
is there an implicit argument that's not erased here, i.e. a hidden #hsz ?

Reply:
I don't think so. It just complained about state and ghost not being applicable at the same time, so I copied this trick from your previous changes. `hsz` is not erased (that's the new state argument in the regional), but `#hsz'` is correctly erased.